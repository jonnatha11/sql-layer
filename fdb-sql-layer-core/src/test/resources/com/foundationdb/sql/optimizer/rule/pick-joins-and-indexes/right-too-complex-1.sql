-- Note: the orders.order_date <> '2011-01-01' cannot be pushed down to the index.
-- For any row with order_date = '2011-01-01' this should return that row once, with null
-- for the customers part, regardless of the contents of the customers table. Pushing it
-- down to the scan would cause it to not be returned at all.
SELECT name, order_date
  FROM customers 
 RIGHT JOIN orders ON customers.cid = orders.cid AND orders.order_date <> '2011-01-01'