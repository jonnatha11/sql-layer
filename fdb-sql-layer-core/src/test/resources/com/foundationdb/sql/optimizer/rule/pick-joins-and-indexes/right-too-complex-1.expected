SelectQuery@615e7597
  ResultSet@7a3e72[name, order_date]
    Project@5999ae9c[customers.name, orders.order_date]
      Select@7896b1b8[]
        JoinNode@6d6de4e1(LEFT/NESTED_LOOPS[orders.order_date != 2011-01-01])
          TableGroupJoinTree@1cf7cbd1(TableGroup@50661532(customers), orders - SingleIndexScan@13f3589(Index(test.orders.order_date[IndexColumn(order_date)]), covering/NONE, rows = 1000, cost = 702.778))
            TableSource@3f0dbef1(orders - TableGroup@50661532(customers))
          TableGroupJoinTree@5df9aeda(TableGroup@50661532(customers), customers - GroupLoopScan@37f22bf(orders - customers, rows = 1, cost = 7.01160))
            TableSource@72b0f2b2(customers - TableGroup@50661532(customers))
